src/transactionsFactories/tokenManagementTransactionIntentsFactory.spec.ts

```diff
@@ -218,4 +219,158 @@ describe("test token management transactions factory", () => {
+        assert.equal(transaction.value, 0n);
+        assert.equal(transaction.gasLimit, 60114500n);
+    });
+
+    it("should create 'Transaction' for updating metadata", () => {
+        const transaction = tokenManagementFactory.createTransactionForUpdatingMetadata({
+            sender: grace.address,
+            tokenIdentifier: "TEST-123456",
+            tokenNonce: 1n,
+            tokenName: "Test",
+            royalties: 1234n,
+            hash: "abba",
+            attributes: Buffer.from("test"),
+            uris: ["firstURI", "secondURI"],
+        });
+
+        assert.deepEqual(
+            transaction.data,
+            Buffer.from(
+                "ESDTMetaDataUpdate@544553542d313233343536@01@54657374@04d2@61626261@74657374@6669727374555249@7365636f6e64555249",
+            ),
+        );
+        assert.equal(transaction.sender, grace.address.toString());
+        assert.equal(transaction.receiver, grace.address.toString());
+        assert.equal(transaction.value, 0n);
+        assert.equal(transaction.gasLimit, 60218000n);
+    });
+
+    it("should create 'Transaction' for recreating metadata", () => {
+        const transaction = tokenManagementFactory.createTransactionForNftMetadataRecreate({
+            sender: grace.address,
+            tokenIdentifier: "TEST-123456",
+            tokenNonce: 1n,
+            tokenName: "Test",
+            royalties: 1234n,
+            hash: "abba",
+            attributes: Buffer.from("test"),
+            uris: ["firstURI", "secondURI"],
+        });
+
+        assert.deepEqual(
+            transaction.data,
+            Buffer.from(
+                "ESDTMetaDataRecreate@544553542d313233343536@01@54657374@04d2@61626261@74657374@6669727374555249@7365636f6e64555249",
+            ),
+        );
+        assert.equal(transaction.sender, grace.address.toString());
+        assert.equal(transaction.receiver, grace.address.toString());
+        assert.equal(transaction.value, 0n);
+        assert.equal(transaction.gasLimit, 60221000n);
+    });
+
+    it("should create 'Transaction' for changing to dynamic", () => {
+        const transaction = tokenManagementFactory.createTransactionForMakingTokenDynamic({
+            sender: grace.address,
+            tokenIdentifier: "TEST-123456",
+        });
+
+        assert.deepEqual(transaction.data, Buffer.from("changeToDynamic@544553542d313233343536"));
+        assert.equal(transaction.sender, grace.address.toString());
+        assert.equal(transaction.receiver, Address.newFromHex(ESDT_CONTRACT_ADDRESS_HEX, config.addressHrp).toBech32());
+        assert.equal(transaction.value, 0n);
+        assert.equal(transaction.gasLimit, 60107000n);
+    });
+
+    it("should create 'Transaction' for updating token id", () => {
+        const transaction = tokenManagementFactory.createTransactionForUpdatingTokenId({
+            sender: grace.address,
+            tokenIdentifier: "TEST-123456",
```

Comment:
I assume the ticker remains, but the random part is being regenerated?

Reply:
the name is a bit misleading, but the token's identifier does not change. From the docs:
```
The token type can be updated to the lastest version, which will update token type and propagate it to shard's
system account. Currently, token type is correctly saved only on metachain and there is no type related information
on shard level, the shard only knows if the token is non fungible with implicit type `NonFungibleESDT`, but it
does not know specifically if it's `NonFungibleESDT`, `MetaESDT` or `SemiFungibleESDT`. So, the update operation will
proceed in the following way:
- if token type is `NonFungibleESDT` it will be set to `NonFungibleESDTv2`, and it will be propagated to shard's system account
- if token type is `MetaESDT` or `SemiFungibleESDT` it will be propagated to shard's system account
```